sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_client.py

```diff
@@ -112,34 +112,34 @@ def __init__(self, key, credential, **kwargs):
         self._key_id = None  # type: Optional[KeyVaultResourceId]
 
         if isinstance(key, KeyVaultKey):
-            self._key = key.key
+            self._key = key.key  # type: Union[JsonWebKey, KeyVaultKey, str, None]
             self._key_id = parse_key_vault_id(key.id)
             if key.properties._attributes:  # pylint:disable=protected-access
                 self._not_before = key.properties.not_before
                 self._expires_on = key.properties.expires_on
         elif isinstance(key, six.string_types):
             self._key = None
             self._key_id = parse_key_vault_id(key)
-            self._keys_get_forbidden = None  # type: Optional[bool]
+            self._keys_get_forbidden = False
         elif self._jwk:
             self._key = key
         else:
             raise ValueError("'key' must be a KeyVaultKey instance or a key ID string including a version")
 
-        if not (self._jwk or self._key_id.version):
+        if not (self._jwk or self._key_id.version if self._key_id else None):
             raise ValueError("'key' must include a version")
 
         if self._jwk:
             try:
-                self._local_provider = get_local_cryptography_provider(self._key)
+                self._local_provider = get_local_cryptography_provider(cast(JsonWebKey, self._key))
                 self._initialized = True
             except Exception as ex:  # pylint:disable=broad-except
                 six.raise_from(ValueError("The provided jwk is not valid for local cryptography"), ex)
         else:
             self._local_provider = NoLocalCryptography()
             self._initialized = False
 
-        self._vault_url = None if self._jwk else self._key_id.vault_url
+        self._vault_url = None if self._jwk else self._key_id.vault_url if self._key_id else None  # type: ignore
```

Comment:
nit: since we make `self._vault_url` in multiple cases here, I think reordering the evaluation a bit makes things clearer:
```suggestion
        self._vault_url = None if (self._jwk or self._key_id is None) else self._key_id.vault_url  # type: ignore
```